common/persistence/cassandra/cassandraQueue.go

```diff
@@ -187,3 +191,68 @@ func getMessagePayload(message map[string]interface{}) []byte {
 func getMessageID(message map[string]interface{}) int {
 	return message["message_id"].(int)
 }
+
+func (q *cassandraQueue) UpdateAckLevel(messageID int, clusterName string) error {
+	version := 0
+	clusterAckLevels := map[string]int{clusterName: messageID}
+	query := q.session.Query(templateInsertQueueMetadataQuery, q.queueType, clusterAckLevels, version)
```

Comment:
plz do not put business logic in the persistence layer.

for initialization of the record, plz put it in cassandraQueue initialization logic (persistence layer)

Reply:
the logic here is basically assuming that the record does not exists and insert an initial version of record.
this is 99.99% of the case, which means that this line of conditional insert will be used as a get method.

conditional query on Cassandra is based on LWT, which will take more network latency compared to a simple get method

Reply:
synced with @meiliang86, there are couple of differences in Cassandra and SQL in terms of implementation.

overall, the persistence layer will expose APIs which hides all DB specific logic. 
Cassandra & SQL will each has DB specific logic layer doing conversion

Reply:
Will do the following:
- Queue interface will accept deltas (i.e. individual cluster name and ack level)
- Cassandra will do CAS, and we will have two separate methods to do that (get, update)
- SQL will do select for update and then update, which needs to happen in one transaction.

Logically both requires a get and update method. However we cannot have a general get and update interface for the queue because we don't want to leak the transaction concept out.